src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -781,6 +781,23 @@ protected theorem extractLsb_ofNat (x n : Nat) (hi lo : Nat) :
     (extractLsb' start len x).getLsbD i = (i < len && x.getLsbD (start+i)) := by
   simp [getLsbD, Nat.lt_succ]
 
+@[simp] theorem getMsbD_extractLsb' {start len : Nat} {x : BitVec w} {i : Nat} :
+    (extractLsb' start len x).getMsbD i =
+    (decide (i < len) &&
+    (decide (start + (len - 1 - i) < w) && x.getMsbD (w - 1 - (start + (len - 1 - i))))) := by
+  rcases len with rfl | len
+  · simp
+  · have h : (len - i < len + 1) := by omega
+    rw [getMsbD_eq_getLsbD, getLsbD_extractLsb', getLsbD_eq_getMsbD]
+    simp [h]
+
+@[simp] theorem msb_extractLsb' {start len : Nat} {x : BitVec w} :
+    (extractLsb' start len x).msb =
+    (decide (0 < len) && x.getLsbD (start + len - 1)) := by
+  rcases len with rfl | len
```

Comment:
```suggestion
  cases len
```

There's nothing wrong with using rcases here, per se, but we don't even refer to `len` after so regular `cases` does the job just fine